src/analyser.cpp

```diff
@@ -0,0 +1,1509 @@
+                        Type::ALGEBRAIC;
+
+            mVariable = variable;
+
+            relevantCheck = true;
+        }
+    }
+
+    return relevantCheck;
+}
+
+/**
+ * @brief The Analyser::AnalyserImpl struct.
+ *
+ * The private implementation for the Analyser class.
+ */
+struct Analyser::AnalyserImpl
+{
+    Analyser *mAnalyser = nullptr;
+
+    AnalyserModelPtr mModel = nullptr;
+
+    std::vector<AnalyserInternalVariablePtr> mInternalVariables;
+    std::vector<AnalyserInternalEquationPtr> mInternalEquations;
+
+    explicit AnalyserImpl(Analyser *analyser);
+
+    static bool compareVariablesByName(const AnalyserInternalVariablePtr &variable1,
+                                       const AnalyserInternalVariablePtr &variable2);
+
+    static bool isStateVariable(const AnalyserInternalVariablePtr &variable);
+    static bool isConstantOrAlgebraicVariable(const AnalyserInternalVariablePtr &variable);
+
+    static bool compareVariablesByTypeAndIndex(const AnalyserInternalVariablePtr &variable1,
+                                               const AnalyserInternalVariablePtr &variable2);
+
+    static bool compareEquationsByVariable(const AnalyserInternalEquationPtr &equation1,
+                                           const AnalyserInternalEquationPtr &equation2);
+
+    size_t mathmlChildCount(const XmlNodePtr &node) const;
+    XmlNodePtr mathmlChildNode(const XmlNodePtr &node, size_t index) const;
+
+    AnalyserInternalVariablePtr analyserVariable(const VariablePtr &variable);
+
+    VariablePtr voiFirstOccurrence(const VariablePtr &variable,
+                                   const ComponentPtr &component);
+
+    void processNode(const XmlNodePtr &node, AnalyserEquationAstPtr &ast,
+                     const AnalyserEquationAstPtr &astParent,
+                     const ComponentPtr &component,
+                     const AnalyserInternalEquationPtr &equation);
+    void processComponent(const ComponentPtr &component);
+
+    void doEquivalentVariables(const VariablePtr &variable,
+                               std::vector<VariablePtr> &equivalentVariables) const;
+    std::vector<VariablePtr> equivalentVariables(const VariablePtr &variable) const;
+
+    void processEquationAst(const AnalyserEquationAstPtr &ast);
+
+    double scalingFactor(const VariablePtr &variable);
+
+    void scaleAst(const AnalyserEquationAstPtr &ast,
+                  const AnalyserEquationAstPtr &astParent,
+                  double scalingFactor);
+    void scaleEquationAst(const AnalyserEquationAstPtr &ast);
+
+    void processModel(const ModelPtr &model);
+};
+
+Analyser::AnalyserImpl::AnalyserImpl(Analyser *analyser)
+    : mAnalyser(analyser)
+    , mModel(std::shared_ptr<AnalyserModel> {new AnalyserModel {}})
+{
+}
+
+bool Analyser::AnalyserImpl::compareVariablesByName(const AnalyserInternalVariablePtr &variable1,
+                                                    const AnalyserInternalVariablePtr &variable2)
```

Comment:
This function doesn't just compare variable names, also component names?  Should the name reflect that?

Reply:
It's still comparing names, so I am happy with that. Also, what would be an alternative name for that function? `compareVariablesByComponentAndVariableName`? Personally, I find `compareVariablesByName` exhaustive enough.